### PR TITLE
fixes#2999 - Added first timer introduction

### PR DESCRIPTION
**Pull request by devansh-299**
## Description

I used this library (https://github.com/mreram/ShowCaseView) to implement the first timer's introduction to the specified view

 ## Screen-Recording 
 

GIF-200116_215212

 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [x] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by teolemon** [comment]
Thank you very much for the PR @devansh-299 👍 
There was an issue uploading the screenshot.

**Comment by teolemon** [comment]
@deniger can you have a look at this one ? It's an onboarding for the image selector.

**Comment by devansh-299** [comment]
> Thank you very much for the PR @devansh-299 
> There was an issue uploading the screensho

> @deniger can you have a look at this one ? It's an onboarding for the image selector.

Yeah I am updating the PR😅

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductImageManagementActivity.java [hunk 1]
```diff
@@ -125,6 +133,37 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         onRefresh(false);
     }
 
+    private void ShowCaseIntro(String title, String content, int viewId, final int type) {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductImageManagementActivity.java]: Follow methods naming style
```suggestion
    private void startShowcase(String title, String content, int viewId, final int type) {
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductImageManagementActivity.java [hunk 2]
```diff
@@ -125,6 +133,37 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
...
+        new GuideView.Builder(this)
+            .setTitle(title)
+            .setContentText(content)
+            .setTargetView(findViewById(viewId))
+            .setContentTextSize(12)
+            .setTitleTextSize(16)
+            .setDismissType(GuideView.DismissType.outside)
+            .setGuideListener(view -> {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ProductImageManagementActivity.java]: Wouldn't a switch case be cleaner?

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -1018,4 +1018,21 @@
     <string name="load_ingredient_detection_data_summary">No ingredient detection data has been found, try to load it from server</string>
     <string name="white_check_image_descr">A white check</string>
     <string name="hint_packaging">Examples: plastic, paper, glass</string>
+    <string name="check_first_time">check_first_time</string>
+
+    <!--ShowcaseView Strings-->
+    <string name="title_image_type">Select Image Type</string>
+    <string name="content_image_type">Use this to select the different image types</string>
```
Comment (VaiTon) [inline app/src/main/res/values/strings.xml]: I would change "Use this" with "Press here", or something more specific than "use".

**Comment by devansh-299** [comment]
@VaiTon I have updated the PR. Thanks for the suggestions :)

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1018,4 +1018,21 @@
...
+    <string name="title_image_type">Select Image Type</string>
+    <string name="content_image_type">Select image type from here</string>
+    <string name="title_choose_language">Select Language</string>
+    <string name="content_choose_language">Select the product language</string>
+    <string name="title_add_photo">Add Photo</string>
+    <string name="content_add_photo">Press here to add photos</string>
+    <string name="title_choose_photo">Choose Photo</string>
+    <string name="content_choose_photo">Press here to choose photo</string>
+    <string name="title_edit_photo">Edit Photo</string>
+    <string name="content_edit_photo">Press here to edit the photo</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_edit_photo">Edit, crop and rotate the photo</string>
```

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1018,4 +1018,21 @@
...
+
+    <!--ShowcaseView Strings-->
+    <string name="title_image_type">Select Image Type</string>
+    <string name="content_image_type">Select image type from here</string>
+    <string name="title_choose_language">Select Language</string>
+    <string name="content_choose_language">Select the product language</string>
+    <string name="title_add_photo">Add Photo</string>
+    <string name="content_add_photo">Press here to add photos</string>
+    <string name="title_choose_photo">Choose Photo</string>
+    <string name="content_choose_photo">Press here to choose photo</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_choose_photo">Choose an existing photo from other contributors</string>
```

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -1018,4 +1018,21 @@
...
     <string name="hint_packaging">Examples: plastic, paper, glass</string>
+    <string name="check_first_time">check_first_time</string>
+
+    <!--ShowcaseView Strings-->
+    <string name="title_image_type">Select Image Type</string>
+    <string name="content_image_type">Select image type from here</string>
+    <string name="title_choose_language">Select Language</string>
+    <string name="content_choose_language">Select the product language</string>
+    <string name="title_add_photo">Add Photo</string>
+    <string name="content_add_photo">Press here to add photos</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_add_photo">Press here to take a new photo</string>
```

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1018,4 +1018,21 @@
     <string name="load_ingredient_detection_data_summary">No ingredient detection data has been found, try to load it from server</string>
     <string name="white_check_image_descr">A white check</string>
     <string name="hint_packaging">Examples: plastic, paper, glass</string>
+    <string name="check_first_time">check_first_time</string>
+
+    <!--ShowcaseView Strings-->
+    <string name="title_image_type">Select Image Type</string>
+    <string name="content_image_type">Select image type from here</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_image_type">Choose the photo to work on (front, ingredients or nutrition)</string>
```

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -1018,4 +1018,21 @@
...
+    <string name="title_choose_language">Select Language</string>
+    <string name="content_choose_language">Select the product language</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_choose_language">Select the language of the photo</string>
```

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1018,4 +1018,21 @@
...
+    <string name="title_add_photo">Add Photo</string>
+    <string name="content_add_photo">Press here to add photos</string>
+    <string name="title_choose_photo">Choose Photo</string>
+    <string name="content_choose_photo">Press here to choose photo</string>
+    <string name="title_edit_photo">Edit Photo</string>
+    <string name="content_edit_photo">Press here to edit the photo</string>
+    <string name="title_unselect_photo">Unselect Photo</string>
+    <string name="content_unselect_photo">Press here to unselect current photo</string>
+    <string name="title_exit">Exit</string>
+    <string name="content_exit">Press here to exit</string>
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="content_exit">Press here to return to the product</string>
```

**Comment by devansh-299** [comment]
@teolemon I have updated the PR, thanks for the review

**Comment by teolemon** [comment]
Thanks a lot @devansh-299 and congratulations on your first PR :-)

**Pull request merged**